pkg/controller/attach_detach.go

```diff
@@ -157,7 +163,38 @@ func (r *ReconcileAttachDetach) triggerAttach(ctx context.Context, azVolumeAttac
 		cloudCtx, cloudCancel := context.WithTimeout(context.Background(), cloudTimeout)
 		defer cloudCancel()
 		updateCtx := context.Background()
+		// attempt to attach the disk to a node
 		response, attachErr := r.attachVolume(cloudCtx, azVolumeAttachment.Spec.VolumeID, azVolumeAttachment.Spec.NodeName, azVolumeAttachment.Spec.VolumeContext)
+		// if the disk is attached to a different node
+		if _, ok := attachErr.(*volerr.DanglingAttachError); ok {
```

Comment:
Instead of ignore the result of the type assertion, consider storing it in a variable (e.g. `danglingAttachErr`) to reuse later instead of using a type assertion like on line 171.